### PR TITLE
Bugfix/encoding for position files

### PR DESCRIPTION
**Pull request by mhdkassir**
## Summary
Adds a new method to read non-csv text files with an encoding different than UTF-8

**Related Issue: #289**

**PR Type: Bug Fix**

**Status: Ready to review**

**Breaking change? No**

### Solution
`PositionIngestinJob.loadDataSet() `checks for schema encoding before reading the data. When the encoding is different than UTF-8, data is loaded using the method `PositionIngestionUtil.loadDfWithEncoding` which can take in consideration the specified encoding.

### How has this been tested?
Adding a new unit test

### Other changes
minor correction in Domain validation error message

## Contributor checklist:
- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/job/ingest/PositionIngestionJob.scala [hunk 1]
```diff
@@ -57,8 +59,14 @@ class PositionIngestionJob(
     */
   override def loadDataSet(): Try[DataFrame] = {
     try {
-      val df =
-        session.read.option("encoding", metadata.getEncoding()).text(path.map(_.toString): _*)
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/ingest/PositionIngestionJob.scala:61]: I don't understand why setting the option to "ISO-8859-1" in the encoding property won't load the file in the specified encoding ? Is it a Spark bug ?

Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/ingest/PositionIngestionJob.scala:61]: Looking ag this PR https://github.com/apache/spark/pull/21287
I feel like there is an easier way too support custom encoding in non csv text file

Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/ingest/PositionIngestionJob.scala:61]: unfortunately this PR was marked as Won't Fix

Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/job/ingest/PositionIngestionJob.scala:61]: > unfortunately this PR was marked as Won't Fix

exactly, I found another PR on spark about this subject that I indicated in the bug description (https://issues.apache.org/jira/browse/SPARK-1849) and it was also marked as _Won't Fix_

**Pull request merged**